### PR TITLE
Reworking EventListener v.1

### PR DESCRIPTION
**Pull request by RedstoneFuture**
**Renaming "e" variable to "event" in all EventHandlers**

**Reworked (Refactoring, Improvement the checks, Continuity check) and tested event listener Ingame**

- PlayerRespawnEvent: EL GL LL
- PlayerDeathEvent: EL GL
- InventoryClickEvent: LL
- EntityDamageEvent: LL
- ServerListPingEvent: PL
- FoodLevelChangeEvent: PL
- BlockPlaceEvent: PL
- PlayerDropItemEvent: PL
- PlayerPickupItemEvent: PL

---

EL: EndListener
GL: GameListener
LL: LobbyListener
PL: PlayerListener

### CODE REVIEW

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java [hunk 1]
```diff
@@ -153,6 +157,34 @@ public void onLeaveWithStuff(PlayerArenaLeaveEvent event) {
         MissileWars.getInstance().getSignRepository().getSigns(game).forEach(MWSign::update);
     }
 
+    // cancel anvil interaction to block renaming, and thus changing missiles
+    @EventHandler
+    public void onClick(InventoryClickEvent event) {
+        Game game = getGame(event.getWhoClicked().getLocation());
+        if (game != null) {
+            if (event.getInventory().getType() == InventoryType.ANVIL) {
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java:165]: Also use guard clauses, just as in all other events

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java [hunk 2]
```diff
@@ -153,6 +157,34 @@ public void onLeaveWithStuff(PlayerArenaLeaveEvent event) {
...
+                event.setCancelled(true);
+            }
+        }
+    }
+
+    // Internal stuff
+
+    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
+    public void onMaxPlayers(PrePlayerArenaJoinEvent event) {
+        if (event.getGame().getPlayers().size() >= event.getGame().getLobby().getMaxSize() && event.getGame().getState() == GameState.LOBBY)
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java:175]: Simplify if clause and use guard clause

**Review by daniel-naegele — CHANGES_REQUESTED** [review]

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java (hunk 2 above)
Comment (RedstoneFuture) [reply missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java:175]: I down't reworked this event in this PR. Only the "event" variable. The rework will come with one of the next PR.

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java (hunk 1 above)
Comment (RedstoneFuture) [reply missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java:165]: I down't reworked this event in the PlayerListener. Only the "event" variable. The rework will come with one of the next PR.

**Pull request merged**